lib/fabricio/client/client.rb

```diff
@@ -54,10 +55,11 @@ def initialize(options =
       @auth_client = Fabricio::Authorization::AuthorizationClient.new
       session = obtain_session
       network_client = Fabricio::Networking::NetworkClient.new(@auth_client, @session_storage)
+      organization_id_provider = Fabricio::Networking::OrganizationIdProvider.new(lambda { return @app_service.all })
```

Comment:
I'm not 100% happy with this approach, but it's the easiest way I found to break the circular reference between `OrganizationIdProvider` and `AppService` without giving up constructor injection (the provider needs the service to lazily request the list of applications, and the service needs the provider to build some URLs via its model factory).

Reply:
Looks ok to me